src/main/java/graphql/schema/idl/SchemaPrinter.java

```diff
@@ -722,115 +785,104 @@ private void printType(PrintWriter out, GraphQLType type, GraphqlFieldVisibility
+        if (isNullOrEmpty(descriptionText)) {
             return;
         }
 
-        if (descriptionAndComments.comments != null) {
-            descriptionAndComments.comments.forEach(cmt -> {
-                String commentText = cmt.getContent() == null ? "" : cmt.getContent();
-                // it possible that in fact they manage to sneak in a multi line comment
-                // into what should be a single line comment.  So cater for that.
-                List<String> lines = Arrays.asList(commentText.split("\n"));
-                lines.forEach(t -> {
-                    out.write(prefix);
-                    out.write("#");
-                    out.write(commentText);
-                    out.write("\n");
-                });
-            });
-        } else {
-            String runtimeDescription = descriptionAndComments.runtimeDescription;
-            if (!isNullOrEmpty(runtimeDescription)) {
-                Stream<String> stream = Arrays.stream(runtimeDescription.split("\n"));
-                stream.map(s -> prefix + "#" + s + "\n").forEach(out::write);
+        if (!isNullOrEmpty(descriptionText)) {
+            List<String> lines = Arrays.asList(descriptionText.split("\n"));
+            if (options.isDescriptionsAsHashComments()) {
+                printMultiLineHashDescription(out, prefix, lines);
+            } else {
+                if (lines.size() > 1) {
+                    printMultiLineDescription(out, prefix, lines);
+                } else {
+                    printSingleLineDescription(out, prefix, lines.get(0));
+                }
             }
         }
     }
 
-    static class AstDescriptionAndComments {
-
-        String runtimeDescription;
+    private void printMultiLineHashDescription(PrintWriter out, String prefix, List<String> lines) {
+        lines.forEach(l -> out.printf("%s#%s\n", prefix, l));
+    }
 
-        Description descriptionAst;
+    private void printMultiLineDescription(PrintWriter out, String prefix, List<String> lines) {
+        out.printf("%s\"\"\"\n", prefix);
+        lines.forEach(l -> out.printf("%s%s\n", prefix, l));
+        out.printf("%s\"\"\"\n", prefix);
+    }
 
-        List<Comment> comments;
+    private void printSingleLineDescription(PrintWriter out, String prefix, String s) {
+        out.printf("%s\"%s\"\n", prefix, s);
+    }
 
-        public AstDescriptionAndComments(String runtimeDescription, Description descriptionAst, List<Comment> comments) {
-            this.runtimeDescription = runtimeDescription;
-            this.descriptionAst = descriptionAst;
-            this.comments = comments;
-        }
+    private boolean hasDescription(Object descriptionHolder) {
+        String description = getDescription(descriptionHolder);
+        return !isNullOrEmpty(description);
     }
 
-    private AstDescriptionAndComments getDescriptionAndComments(Object descriptionHolder) {
+    private String getDescription(Object descriptionHolder) {
```

Comment:
Can this be a static function? does it need references to this?

Reply:
could be - but almost no other method is static in this class - so its consistent in that regard